src/plugins/es_ui_shared/static/forms/hook_form_lib/components/use_field.tsx

```diff
@@ -49,53 +49,95 @@ export interface Props<T, FormType = FormData, I = T> {
 }
 
 function UseFieldComp<T = unknown, FormType = FormData, I = T>(props: Props<T, FormType, I>) {
+  const form = useFormContext<FormType>();
+  const { getFieldDefaultValue, __readFieldConfigFromSchema, __updateDefaultValueAt } = form;
+
   const {
     path,
-    config,
+    config = __readFieldConfigFromSchema<T, FormType, I>(props.path),
     defaultValue,
     component,
     componentProps,
     readDefaultValueOnForm = true,
     onChange,
     onError,
     children,
-    validationData: customValidationData,
-    validationDataProvider: customValidationDataProvider,
+    validationData,
+    validationDataProvider,
     ...rest
   } = props;
 
-  const form = useFormContext<FormType>();
   const ComponentToRender = component ?? 'input';
   const propsToForward = { ...componentProps, ...rest };
 
-  const fieldConfig: FieldConfig<T, FormType, I> & { initialValue?: T } =
-    config !== undefined
-      ? { ...config }
-      : ({
-          ...form.__readFieldConfigFromSchema(path),
-        } as Partial<FieldConfig<T, FormType, I>>);
+  const initialValue = useMemo<T>(() => {
+    // The initial value of the field.
```

Comment:
Should this information be added to the docs as well?

Reply:
It is already in the docs: https://docs.elastic.dev/form-lib/core/default-value#order-of-precedence